ext/pdo_sqlite/sqlite_driver.c

```diff
@@ -224,8 +224,27 @@ static zend_string* sqlite_handle_quoter(pdo_dbh_t *dbh, const zend_string *unqu
 {
 	char *quoted;
 	if (ZSTR_LEN(unquoted) > (INT_MAX - 3) / 2) {
+		php_error_docref(NULL, E_WARNING, "String is too long to be quoted");
 		return NULL;
 	}
+	// check if it contains a null byte
+	if (memchr(ZSTR_VAL(unquoted), '\0', ZSTR_LEN(unquoted))) {
+		if(((ZSTR_LEN(unquoted) * 2) + 3) > (INT_MAX - 3) / 2) {
```

Comment:
The overflow check is incorrect, leading to a potential integer overflow resulting in a buffer overflow.